broadinstitute/rawls
# Create Azure relay on Workspace creation (WOR-136).

Note that the endpoint has not yet been added to Swagger-- that will be done shortly. For now, to test this you can use curl:

Example for running against local "front" rawls, with a test namespace/workspace name:
````
curl -X POST --location "https://local.broadinstitute.org:20443/api/workspaces/mc" \                                                                  
    -H "accept: application/json" \
    -H "Authorization: Bearer <<output of  'gcloud auth print-access-token'>>" \
    -H "Content-Type: application/json" \
    -H "User-Agent: Testing" \
    -d "{
          \"namespace\": \"car_testing_ns\",
          \"name\": \"car_18\",
          \"attributes\": {},
          \"cloudPlatform\": \"AZURE\", \"region\": \"eastus\"
        }"

````

Deletion of the relay will happen in WOR-228.

  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
